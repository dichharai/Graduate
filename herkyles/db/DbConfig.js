import * as firebase from "firebase";

const firebaseConfig = {
    apiKey: "AIzaSyD5Ha8fSPEBkjqMchNVy8zU6vB-WvcGYTM",
    authDomain: "herkyles-85e8e.firebaseapp.com",
    databaseURL: "https://herkyles-85e8e.firebaseio.com",
    projectId: "herkyles-85e8e",
    storageBucket: "herkyles-85e8e.appspot.com",
    messagingSenderId: "812934741247"
};

export const firebaseApp = firebase.initializeApp(firebaseConfig); 