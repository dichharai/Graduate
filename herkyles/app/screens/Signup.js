import React, { Component } from 'react'; 
import {Button, View, Text, StyleSheet, AsyncStorage} from 'react-native'; 
import {firebaseApp} from '../../db/DbConfig'; 
import {LinkBtns} from '../components/Buttons/LinkBtns';

import {Container} from '../components/Container';
import ErrorStatus from '../components/Status/ErrorStatus';  
import t from 'tcomb-form-native';
import EStyleSheet from 'react-native-extended-stylesheet'; 

const Form = t.form.Form;
const styles = EStyleSheet.create({ 
    container: {
        backgroundColor: '#ffffff', 
        justifyContent: 'center',
        //marginTop: 50, 
        padding: 20,
    },
    status: {
        justifyContent: 'center',
        color: 'red',
    }

}); 
const User = t.struct({
    email: t.String, 
    password: t.String, 
    terms: t.Boolean, 
}); 

const formStyles = {
    ...Form.stylesheet,
    formGroup: {
        normal: {
        marginBottom: 10
        },
    },
    controlLabel: {
        normal: {
        fontSize: 12,
        color: '#7F7D7D',
        fontWeight: '400',
        marginBottom: 7,
        },
        // the style applied when a validation error occurs
        error: {
        color: '#7F7D7D',
        fontSize: 12,
        marginBottom: 7,
        fontWeight: '400'
        }
    }
}; 
const options = {
    fields: {
      email: {
        error: 'Please enter an email',
      },
      password: {
        error: 'Please enter a password',
        password: true, 
        secureTextEntry: true, 
      },
      terms: {
        label: 'Agree to Terms',
      },
    },
    stylesheet: formStyles,
};


class Signup extends Component{
    static navigationOptions = {
        title: "Signup", 
        headerStyle: {
            backgroundColor: "#000000",
        }, 
          headerTintColor: '#facf33',
    };
    state = {status: ''};
    _signUpAsync = async () => {
        try{
            const statusFlag = 'successfully signed up!';
            await AsyncStorage.setItem("signupToLogin",statusFlag);
            let stlStatus = await AsyncStorage.getItem("signupToLogin"); 
            console.log('status from signup ' + stlStatus );
            this.props.navigation.navigate('Login', {});

        }catch(e){
            console.log(e);
        }
    };
    handleSubmit = () => {
        //const value = 'apple';
        const value = this._form.getValue(); // use that ref. to get the form value
        console.log(value);
        //console.log('value', value.email + " " + value.password + " " + value.terms); 
        if (value != null){
            //const {email, password} = [value.email.toString(), value.password.toString()];

            firebaseApp.auth().createUserWithEmailAndPassword(value.email, value.password).then(()=> {
                console.log("successful");
                this._signUpAsync();

            }).catch((error) => {
                // Sign up was not successful
                console.log("sign up was not successful");
                this.setState({status:error.code});
            });
        }
    };
    render(){
        return (
            <Container>
                <Form
                    ref={c=>this._form=c} // assign a ref
                    type={User}
                    options={options}
                />
                <LinkBtns 
                    text='Sign up'
                    onPress={this.handleSubmit}
                />
                <ErrorStatus 
                    text={this.state.status} 
                />
                
            
            </Container>            
        );
    }
};

export default Signup; 