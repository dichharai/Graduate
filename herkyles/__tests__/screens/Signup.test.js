import React from "react";
import Signup from "../../app/screens/Signup";

import renderer from "react-test-renderer";
import * as firebase from '../../db/DbConfig'

describe ("Signup", () => {
  it("renders without crashing", () => {
    const rendered = renderer.create(<Signup />).toJSON();
    expect(rendered).toBeTruthy();
    expect(rendered).toMatchSnapshot();
  });
  
});
/*
describe('handleSubmit', () => {
  beforeAll(() => {
    firebase.firebase.auth = jest.fn().mockReturnValue({
    })
  })
});
*/