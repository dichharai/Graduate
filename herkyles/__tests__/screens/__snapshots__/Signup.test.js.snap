// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup renders without crashing 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={undefined}
  testID={undefined}
>
  <View
    style={Object {}}
  >
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#7F7D7D",
            "fontSize": 12,
            "fontWeight": "400",
            "marginBottom": 7,
          }
        }
      >
        Email
      </Text>
      <View
        style={Object {}}
      >
        <TextInput
          accessibilityLabel="Email"
          allowFontScaling={true}
          autoCapitalize={undefined}
          autoCorrect={undefined}
          autoFocus={undefined}
          blurOnSubmit={undefined}
          clearButtonMode={undefined}
          clearTextOnFocus={undefined}
          editable={undefined}
          enablesReturnKeyAutomatically={undefined}
          keyboardAppearance={undefined}
          keyboardType={undefined}
          maxLength={undefined}
          multiline={undefined}
          numberOfLines={undefined}
          onBlur={undefined}
          onChange={undefined}
          onChangeText={[Function]}
          onContentSizeChange={undefined}
          onEndEditing={undefined}
          onFocus={undefined}
          onKeyPress={undefined}
          onLayout={undefined}
          onSelectionChange={undefined}
          onSubmitEditing={undefined}
          placeholder={undefined}
          placeholderTextColor={undefined}
          returnKeyType={undefined}
          secureTextEntry={undefined}
          selectTextOnFocus={undefined}
          selectionColor={undefined}
          selectionState={undefined}
          style={
            Object {
              "borderColor": "#cccccc",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "#000000",
              "fontSize": 17,
              "height": 36,
              "marginBottom": 5,
              "paddingHorizontal": 7,
              "paddingVertical": 7,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#7F7D7D",
            "fontSize": 12,
            "fontWeight": "400",
            "marginBottom": 7,
          }
        }
      >
        Password
      </Text>
      <View
        style={Object {}}
      >
        <TextInput
          accessibilityLabel="Password"
          allowFontScaling={true}
          autoCapitalize={undefined}
          autoCorrect={undefined}
          autoFocus={undefined}
          blurOnSubmit={undefined}
          clearButtonMode={undefined}
          clearTextOnFocus={undefined}
          editable={undefined}
          enablesReturnKeyAutomatically={undefined}
          keyboardAppearance={undefined}
          keyboardType={undefined}
          maxLength={undefined}
          multiline={undefined}
          numberOfLines={undefined}
          onBlur={undefined}
          onChange={undefined}
          onChangeText={[Function]}
          onContentSizeChange={undefined}
          onEndEditing={undefined}
          onFocus={undefined}
          onKeyPress={undefined}
          onLayout={undefined}
          onSelectionChange={undefined}
          onSubmitEditing={undefined}
          placeholder={undefined}
          placeholderTextColor={undefined}
          returnKeyType={undefined}
          secureTextEntry={true}
          selectTextOnFocus={undefined}
          selectionColor={undefined}
          selectionState={undefined}
          style={
            Object {
              "borderColor": "#cccccc",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "#000000",
              "fontSize": 17,
              "height": 36,
              "marginBottom": 5,
              "paddingHorizontal": 7,
              "paddingVertical": 7,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#7F7D7D",
            "fontSize": 12,
            "fontWeight": "400",
            "marginBottom": 7,
          }
        }
      >
        Agree to Terms
      </Text>
      <RCTSwitch
        accessibilityLabel="Agree to Terms"
        disabled={false}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor={undefined}
        onValueChange={[Function]}
        style={
          Array [
            Object {
              "height": 31,
              "width": 51,
            },
            Object {
              "marginBottom": 4,
            },
          ]
        }
        thumbTintColor={undefined}
        tintColor={undefined}
        value={false}
      />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={undefined}
      >
        Sign up
      </Text>
    </View>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={undefined}
    >
      
    </Text>
  </View>
</View>
`;
